### DISCUSSION TITLE
Changing beta channel prior distributions from weakly to informative

### DISCUSSION THREAD

**Discussion by BahgatN** [pymc-labs/pymc-marketing#588] [Q&A] [answered]
I'm relatively new to pymc_marketing, and I've observed a significant decrease in sampling algorithm speed when I switched the beta media channel priors from Halfnormal to a beta distribution. Can someone please shed light on why this might be happening?

**Comment by williambdean** [comment]
Hello @BahgatN 

What are your parameters estimates when the prior is HalfNormal? Is much of there density >1?
Restricting the domain to (0, 1) might impact the MCMC 
How are the sampling diagnostics before and after?

**Reply by BahgatN** [reply]
Hi @wd60622 

The estimates show notable differences between the models using beta distribution compared to HalfNormal. This discrepancy arises because I re-parameterized the distribution to incorporate highly informative priors, which align with my beliefs.

In both models there are convergence and the trace plots for all parameters are good, but with beta priors, I had to adjust the tune size and acceptance rate to prevent algorithm divergence

**Reply by williambdean** [reply]
What do you mean by notably different? Can you through out ball park values? Is the distribution mainly above 1?

How does the sampling perform if you set the prior to distribution with (0, \inf) domain but with most of the prior mass between 0 and 1. 
(This could be gamma prior and found initially `pm.find_constrained_prior`)

**Reply by BahgatN** [reply]
@wd60622  
The shift to Beta priors from HalfNormal resulted in notably different means for the contribution of each media channel. The Beta distribution, bounded between 0 and 1, facilitates informative priors and is well-suited for pymc_marketing package, especially due to its utilization of the MaxAbs Scalar.
See [How can i use prior to specific the beta parameter of a channel variable?](https://github.com/pymc-labs/pymc-marketing/discussions/511)
I did not get your piont about the zero infinity domain

**Reply by williambdean** [reply, marked as answer]
Based on the information you've provided, I suspect that the sampling is slower due to parameters being more likely outside of the domain of the Beta distribution
